### PR TITLE
[repo] Prepare to .NET9 - reenable analysis level

### PR DESCRIPTION
**Pull request by Kielek**
## Changes

Follow up #2251

All leftovers. Enables `<AnalysisLevel>latest-all</AnalysisLevel>`

PR is a bit bigger than usual, but I would like to finish finally this effort. It should be safe to review commits independently.
Most important commits:
* c1f1f3d4da2c9c2e0bc6d9f36e0535dae5fe6441
* 88321b30ce0c2cc5b1559cc3aee517904e3fca5a
* f78668ace2d2a4c1bf295ecfd99929281fedc3ba
* 6c2101c0cc137209314a726ba74adf15ef7ec0f4

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* ~~[ ] Unit tests added/updated~~
* ~~[ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes~~
* ~~[ ] Changes in public API reviewed (if applicable)~~

### CODE REVIEW

**Review by RassK — APPROVED** [review]

**Review by fred2u — APPROVED** [review]

**Review by xsoheilalizadeh — APPROVED** [review]

**Review by ppittle — APPROVED** [review]

File: src/OpenTelemetry.Sampler.AWS/AWSXRayRemoteSamplerBuilder.cs [hunk 1]
```diff
@@ -70,7 +70,9 @@ public AWSXRayRemoteSamplerBuilder SetEndpoint(string endpoint)
     /// <returns>an instance of <see cref="Trace.Sampler"/>.</returns>
     public Trace.Sampler Build()
     {
+#pragma warning disable CA2000
```
Comment (TimothyMothra) [inline src/OpenTelemetry.Sampler.AWS/AWSXRayRemoteSamplerBuilder.cs:73]: nit: I think it's a good idea to include the title of what is being disabled.

```suggestion
#pragma warning disable CA2000 // Dispose objects before losing scope
```

File: test/OpenTelemetry.Contrib.Shared.Tests/SqlProcessorTestCases.cs [hunk 2]
```diff
@@ -28,7 +28,9 @@ public static IEnumerable<object[]> GetTestCases()
         }
     }
 
+#pragma warning disable CA1034
```
Comment (TimothyMothra) [inline test/OpenTelemetry.Contrib.Shared.Tests/SqlProcessorTestCases.cs:31]: nit:

```suggestion
#pragma warning disable CA1034 // Nested types should not be visible
```

**Review by TimothyMothra — APPROVED** [review]
LGTM

**Review by fred2u — APPROVED** [review]

**Comment by Kielek** [comment]
@CodeBlanch, @alanwest - I am merging it to avoid any big conflicts here. Feel free to comment if you have any advices.

**Pull request merged**